core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -132,6 +141,71 @@ public ClientFactoryBuilder workerGroup(EventLoopGroup workerGroup, boolean shut
+     * {@link EventLoop} to handle a connection to the specified {@link Endpoint}.
+     */
+    public ClientFactoryBuilder eventLoopSchedulerFactory(
+            Function<? super EventLoopGroup, ? extends EventLoopScheduler> eventLoopSchedulerFactory) {
+        checkState(maxNumEventLoopsPerHttp1Endpoint == 0 && maxNumEventLoopsPerEndpoint == 0 &&
+                   maxNumEventLoopsFunction == DEFAULT_MAX_NUM_EVENT_LOOPS_FUNCTION,
+                   "Cannot set eventLoopSchedulerFactory when maxEventLoop per endpoint is specified.");
+        this.eventLoopSchedulerFactory = requireNonNull(eventLoopSchedulerFactory, "eventLoopSchedulerFactory");
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of {@link EventLoop}s which will be used to handle HTTP/1.1 connections
+     * except the ones specified by {@link #maxNumEventLoopsFunction(ToIntFunction)}.
+     * {@value DefaultEventLoopScheduler#DEFAULT_MAX_NUM_EVENT_LOOPS} is used by default.
+     */
+    public ClientFactoryBuilder maxNumEventLoopsPerHttp1Endpoint(int maxNumEventLoopsPerEndpoint) {
+        validateMaxNumEventLoopsPerEndpoint(maxNumEventLoopsPerEndpoint);
+        maxNumEventLoopsPerHttp1Endpoint = maxNumEventLoopsPerEndpoint;
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of {@link EventLoop}s which will be used to handle HTTP/2 connections
+     * except the ones specified by {@link #maxNumEventLoopsFunction(ToIntFunction)}.
+     * {@value DefaultEventLoopScheduler#DEFAULT_MAX_NUM_EVENT_LOOPS} is used by default.
+     */
+    public ClientFactoryBuilder maxNumEventLoopsPerEndpoint(int maxNumEventLoopsPerEndpoint) {
+        validateMaxNumEventLoopsPerEndpoint(maxNumEventLoopsPerEndpoint);
+        this.maxNumEventLoopsPerEndpoint = maxNumEventLoopsPerEndpoint;
+        return this;
+    }
+
+    private void validateMaxNumEventLoopsPerEndpoint(int maxNumEventLoopsPerEndpoint) {
+        checkArgument(maxNumEventLoopsPerEndpoint > 0,
+                      "maxNumEventLoopsPerEndpoint: %s (expected: > 0)", maxNumEventLoopsPerEndpoint);
+        checkState(eventLoopSchedulerFactory == null,
+                   "maxNumEventLoopsPerEndpoint() and eventLoopSchedulerFactory() are mutually exclusive.");
+    }
+
+    /**
+     * Sets the {@link ToIntFunction} which takes an {@link Endpoint} and produces the maximum number of
+     * {@link EventLoop}s which will be used to handle connections to the specified {@link Endpoint}.
+     * The function should return {@code -1} for the {@link Endpoint}s which it doesn't want to handle.
+     * For example: <pre>{@code
+     * ToIntFunction<Endpoint> function = endpoint -> {
+     *     if (endpoint.equals(Endpoint.of("foo.com"))) {
+     *         return 5;
+     *     }
+     *     if (endpoint.host().contains("bar.com")) {
+     *         return Integer.MAX_VALUE; // The value will be cut to the size of event loops.
```

Comment:
The value will be clamped at the number of event loops.

Reply:
Fixed PTAL 🙇